js/promises/promises.go

```diff
@@ -28,18 +28,22 @@ import (
 //		    return promise
 //		  }
 func New(vu modules.VU) (p *goja.Promise, resolve func(result any), reject func(reason any)) {
-	p, resolve, reject = vu.Runtime().NewPromise()
+	p, resolveFunc, rejectFunc := vu.Runtime().NewPromise()
 	callback := vu.RegisterCallback()
 
-	return p, func(i interface{}) {
-			callback(func() error {
-				resolve(i)
```

Comment:
So for future reference - the problem here was that `resolve` here  is the same resolve we return - which is the function we in. So this just recurse inside itself.